src/transformers/pipelines/base.py

```diff
@@ -861,7 +861,19 @@ def __init__(
                 raise ValueError(f"{device} unrecognized or not available.")
         else:
             self.device = device if device is not None else -1
+
         self.torch_dtype = torch_dtype
+
+        if not self.torch_dtype and is_torch_available():
+            # If pipeline dtype is not specified, populate it from the model
+            # NB: We should only do this when the extracted dtype is not default one (float32),
+            # because not all models/pipelines support torch_dtype. Here we assume that if the
+            # model dtype is not float32 it is set by the user with torch_dtype param, so the
+            # model or pipeline should support it.
+            model_dtype = getattr(model, "dtype", None) or getattr(model, "dtype", None)
+            if model_dtype not in (torch.float32, "float32", "torch.float32"):
```

Comment:
nit: Relying on *implicit* default dtype of Pytorch may be a bit 
worrisome, but I don't imagine they would change it easily as it should cause more catastrophic issues all over the world.